common/src/main/java/dev/naturecodevoid/voicechatdiscord/audio/AudioCore.java

```diff
@@ -1,47 +1,58 @@
 package dev.naturecodevoid.voicechatdiscord.audio;
 
-import de.maxhenkel.voicechat.api.Player;
 import de.maxhenkel.voicechat.api.Position;
-import de.maxhenkel.voicechat.api.ServerPlayer;
-import dev.naturecodevoid.voicechatdiscord.DiscordBot;
 import dev.naturecodevoid.voicechatdiscord.MathUtil;
 
 import java.util.Iterator;
 import java.util.List;
-import java.util.UUID;
-import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.OptionalInt;
 
 import static dev.naturecodevoid.voicechatdiscord.Common.*;
 
+
+// Utilities and algorithms for operating on audio streams.
 public class AudioCore {
-    /**
-     * Combines multiple audio streams into one.
-     */
-    public static short[] combineAudioStreams(List<short[]> audioStreams) {
-        // https://github.com/DV8FromTheWorld/JDA/blob/11c5bf02a1f4df3372ab68e0ccb4a94d0db368df/src/main/java/net/dv8tion/jda/internal/audio/AudioConnection.java#L529
-        int audioLength = audioStreams.stream().mapToInt(it -> it.length).max().getAsInt();
-        short[] mix = new short[1920]; // 960 PCM samples for each channel
-        int sample;
-        for (int i = 0; i < audioLength; i++) {
-            sample = 0;
-            for (Iterator<short[]> iterator = audioStreams.iterator(); iterator.hasNext(); ) {
-                short[] audio = iterator.next();
-                if (i < audio.length)
-                    sample += audio[i];
+
+    // Number of shorts that are needed in an array to represent 20ms.
+    public static final short COUNT20MS = 960;
+
+    private AudioCore() {throw new UnsupportedOperationException(this.getClass().getSimpleName() + " should not be instantiated.");}
```

Comment:
This shouldn't be needed

We can also just make AudioCore a static class

Reply:
`AudioCore` can't be made a static class because it's not within another class.
I tried to make it more clear that this class doesn't need to be extended nor instantiated, but `final` and `abstract` can't be used together. The best way to enforce both of these (as far as I know) is to make the class `final` and set the initialiser to `private`.